src/main/java/org/broadinstitute/hellbender/utils/pairhmm/DragstrLocus.java

```diff
@@ -0,0 +1,218 @@
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.broadinstitute.hellbender.utils.Nucleotide;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+
+import java.io.*;
+import java.util.Arrays;
+
+public class DragstrLocus {
+
+    private final int chromosomeIndex;
+    private final long start;
+    private final int repeats;
+    private final byte[] unit;
+
+    private DragstrLocus(final int chrIdx, final long start, final byte[] unit, final int repeats) {
+        chromosomeIndex = chrIdx;
+        this.start = start;
+        this.unit = unit;
+        this.repeats = repeats;
+    }
+
+    public static DragstrLocus make(final int chrIdx, final long start, final byte[] unit, final int repeatCount) {
+        ParamUtils.isPositiveOrZero(chrIdx, "chromosome index");
+        ParamUtils.isPositive(start, "start position");
+        final byte[] unitCloned = Utils.nonNull(unit).clone();
+        for (final byte base : unit) {
+            if (!Nucleotide.decode(base).isStandard()) {
+                throw new IllegalArgumentException("bad bases in " + new String(unit));
+            }
+        }
+        ParamUtils.isPositive(repeatCount, "repeat count");
+        return new DragstrLocus(chrIdx, start, unit.clone(), repeatCount);
+    }
+
+    public long getStart() {
+        return start;
+    }
+
+    public long getEnd() {
+        return start + unit.length * repeats - 1;
+    }
+
+    public int getPeriod() {
+        return unit.length;
+    }
+
+    byte[] getUnitUnsafe() {
+        return unit;
+    }
+
+    public byte[] getUnit() {
+        return unit.clone();
+    }
+
+    public int getRepeats() {
+        return repeats;
+    }
+
+    public static BinaryTableWriter<DragstrLocus> binaryWriter(final OutputStream out, final String path) {
+        return new BinaryTableWriter<DragstrLocus>(out, path) {
+            @Override
+            protected void writeRecord(final DragstrLocus record, final DataOutput output) throws IOException {
+                output.writeShort(record.chromosomeIndex);
+                output.writeLong(record.start);
+                output.writeByte(record.unit.length);
+                output.write(record.unit);
+                output.writeByte(record.repeats);
+            }
+        };
+    }
+
+    public static BinaryTableWriter<DragstrLocus> binaryWriter(final File out)
+        throws FileNotFoundException
+    {
+        return binaryWriter(new FileOutputStream(out), out.toString());
+    }
+
+    public static BinaryTableReader<DragstrLocus> binaryReader(final InputStream in) {
```

Comment:
So far as I can tell the only places that these binary readers/writers are used internally for an intermediate output for `SampleSitesForDRAGstrModel` to store the outputs. It seems to me that the binary output formats are fairly heavyweight, perhaps we could investigate reducing our code exposure by simply using gzip output streams instead of these binary streams? They will probably not compress as neatly but will probably save us the risk of bugs in the long run.